create-neon/src/metadata.ts

```diff
@@ -1,16 +1,66 @@
-import { readFile } from "fs/promises";
+import { readFile, writeFile } from "fs/promises";
 
 export interface Metadata {
   project: Project,
   versions: Versions
 }
 
-export interface Project {
+export class Project {
+  private json: any;
+
   name: string;
   version: string;
-  author?: FreeText;
+  author: FreeText | undefined;
   license: string;
-  description?: FreeText;
+  description: FreeText | undefined;
+
+  static async load(source: string, cargoCpArtifact: string): Promise<Project> {
+    return new Project(JSON.parse(await readFile(source, 'utf8')), cargoCpArtifact);
+  }
+
+  constructor(json: any, cargoCpArtifact: string) {
+    this.json = json;
```

Comment:
Why do we have two copies of the data? Overall, I find the class base approach a little complicated. Alternatively,

```ts
interface PackageJson {
    name: string,
    author: string,
    [k: string]: any,
}
```

There might be a package.json types file we could import.